src/IOBuffer.ts

```diff
@@ -268,6 +278,74 @@ export class IOBuffer {
     return bytes;
   }
 
+  /**
+   * Create array of size and TypedArray type
+   * @param size - Size of array
+   * @param word - TypedArray type
+   * @returns
+   */
+  private makeTypedArray(word: Word, size = 1) {
+    switch (word) {
+      case 'int8':
+        return new Int8Array(size);
+      case 'uint8':
+        return new Uint8Array(size);
+      case 'int16':
+        return new Int16Array(size);
+      case 'uint16':
+        return new Uint16Array(size);
+      case 'int32':
+        return new Int32Array(size);
+      case 'uint32':
+        return new Uint32Array(size);
+      case 'float32':
+        return new Float32Array(size);
+      case 'float64':
+        return new Float64Array(size);
+      default:
+        throw new Error('Invalid word');
+    }
+  }
+  /**
+   * Select function for reading multiple values
+   * @param word - number type
+   * @returns - reader function to use elsewhere
+   */
+  private useReader(word: Word) {
+    switch (word) {
+      case 'int8':
+        return this.readInt8.bind(this);
+      case 'uint8':
+        return this.readUint8.bind(this);
+      case 'int16':
+        return this.readInt16.bind(this);
+      case 'uint16':
+        return this.readUint16.bind(this);
+      case 'int32':
+        return this.readInt32.bind(this);
+      case 'uint32':
+        return this.readUint32.bind(this);
+      case 'float32':
+        return this.readFloat32.bind(this);
+      case 'float64':
+        return this.readFloat64.bind(this);
+      default:
+        throw new Error('Invalid word type');
+    }
+  }
+  /**
+   * Makes typed array after reading bytes
+   * @param size - number of elements to read
+   * @param word - type of elements to read
+   */
+  public arrayOf(size: number, word: Word) {
```

Comment:
Please name it `readSomething`, like other read functions.

Reply:
Thanks. 

I was aware there were many silly errors but didn't want to spend much time in case it wasn't useful (plus I don't know enough.) 

I'll try to fix all of that and add more tests.